opentelemetry-api/src/opentelemetry/trace/propagation/textmap.py

```diff
@@ -20,7 +20,64 @@
 TextMapPropagatorT = typing.TypeVar("TextMapPropagatorT")
 
 Setter = typing.Callable[[TextMapPropagatorT, str, str], None]
-Getter = typing.Callable[[TextMapPropagatorT, str], typing.List[str]]
+
+
+def default_get(carrier: TextMapPropagatorT, key: str) -> typing.List[str]:
+    return [carrier.get(key)] if carrier.get(key) else []
+
+
+def default_keys(carrier: TextMapPropagatorT) -> typing.List[str]:
+    return list(carrier.keys())
+
+
+GetterGetFunction = typing.Callable[
+    [TextMapPropagatorT, str], typing.List[str]
+]
+GetterKeysFunction = typing.Callable[[TextMapPropagatorT], typing.List[str]]
+
+
+class Getter:
+    """This class implements a Getter that enables extracting propagated
+    fields from a carrier
+
+    """
+
+    def __init__(
+        self,
+        get: GetterGetFunction[TextMapPropagatorT] = default_get,
+        keys: GetterKeysFunction[TextMapPropagatorT] = default_keys,
+    ):
+        self._get = get
+        self._keys = keys
+
+    def get(self, carrier: TextMapPropagatorT, key: str) -> typing.List[str]:
+        """Function that can retrieve zero
+        or more values from the carrier. In the case that
+        the value does not exist, returns an empty list.
+
+        Args:
+            carrier: and object which contains values that are
+                used to construct a Context. This object
+                must be paired with an appropriate get_from_carrier
+                which understands how to extract a value from it.
+            key: key of a field in carrier.
+        Returns:
+            first value of the propagation key or an empty list if the key doesn't exist.
+        """
+        return self._get(carrier, key)
```

Comment:
I believe `get` is supposed to return `None` if not found. https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/context/api-propagators.md#get

Reply:
@lzchen the existing get_from_carrier method is returning an empty list and most of the implementations of get_from_carrier are using an empty list.
https://github.com/open-telemetry/opentelemetry-python/blob/master/opentelemetry-api/src/opentelemetry/trace/propagation/textmap.py#L51 

Should I still make this change?

Reply:
Yes, I think most instrumentations rely on it returning an empty list but we should change it if the spec says it should return None. I suggest to do that in a separate PR if you want to take it up @nprajilesh

Reply:
@owais Sure, I can pick it up. I would prefer closing this before making that change.

I just have that one typing error remaining in this, If you could help me with that I will be done with this.

Reply:
If you can make an issue for that, that'd be great.

Reply:
https://github.com/open-telemetry/opentelemetry-python/issues/1274

Reply:
@nprajilesh 
Would you like to take this one on?